eng/MultiTargetRoslynComponent.targets.template

```diff
@@ -0,0 +1,41 @@
+<Project>
+  <!-- Reads the version of the compiler APIs that are currently being used in order to pick the correct Roslyn components. -->
+  <Target Name="_{TargetPrefix}ResolveRoslynApiMajorVersion"
+          Condition="'$(_RoslynApiMajorVersion)' == ''">
+    <GetAssemblyIdentity AssemblyFiles="$(RoslynTargetsPath)\Microsoft.Build.Tasks.CodeAnalysis.dll">
+      <Output TaskParameter="Assemblies" ItemName="_CodeAnalysisIdentity" />
+    </GetAssemblyIdentity>
+
+    <PropertyGroup>
+      <_RoslynApiMajorVersion>$([System.Version]::Parse(%(_CodeAnalysisIdentity.Version)).Major)</_RoslynApiMajorVersion>
+    </PropertyGroup>
+  </Target>
+
+  <Target Name="_{TargetPrefix}RoslynComponentMultiTargeting" 
+          Condition="'$(SupportsRoslynComponentVersioning)' != 'true'" 
+          AfterTargets="ResolveAssemblyReferences" 
+          DependsOnTargets="_{TargetPrefix}ResolveRoslynApiMajorVersion">
+    <ItemGroup>
+      <!-- clear _CurrentAnalyzer, in case some other target is using it. -->
+      <_CurrentAnalyzer Remove="@(_CurrentAnalyzer)" />
```

Comment:
Why didn't you just use `{TargetPrefix}` to give this item a namespace?

Reply:
Fixed.